packages/dds/merge-tree/src/test/revertibles.spec.ts

```diff
@@ -44,6 +75,78 @@ describe("MergeTree.Revertibles", () => {
 		logger.validate({ baseText: "123" });
 	});
 
+	it("has reasonable asymptotics in face of remove", () => {
+		const length = 100;
+
+		// track the amount of tracking group linking that occurs
+		let linkCount = 0;
+		let unlinkCount = 0;
+
+		const unspy1 = spyOnMethod(TrackingGroup, "link", () => (linkCount += 1));
+		const unspy2 = spyOnMethod(TrackingGroup, "unlink", () => (unlinkCount += 1));
+		const unspy3 = spyOnMethod(UnorderedTrackingGroup, "link", () => (linkCount += 1));
+		const unspy4 = spyOnMethod(UnorderedTrackingGroup, "unlink", () => (unlinkCount += 1));
+
+		const clients = createClientsAtInitialState(
+			{
+				initialState: "",
+				options: { mergeTreeUseNewLengthCalculations: true },
+			},
+			"A",
+		);
+
+		for (let i = 1; i <= length; i++) {
+			const insertOp = clients.A.insertTextLocal(i - 1, "a");
+			clients.A.applyMsg(
+				clients.A.makeOpMessage(
+					insertOp,
+					/* seq */ i + 1,
+					/* refSeq */ i,
+					clients.A.longClientId,
+					/* minSeq */ 1,
+				),
+			);
+		}
+
+		const driver = createRevertDriver(clients.A);
+
+		const revertibles: MergeTreeDeltaRevertible[] = [];
+		clients.A.on("delta", (_op, delta) => {
+			appendToMergeTreeDeltaRevertibles(driver, delta, revertibles);
+		});
+
+		const op = clients.A.removeRangeLocal(0, length - 1);
+
+		clients.A.applyMsg(
+			clients.A.makeOpMessage(
+				op,
+				/* seq */ length + 1,
+				/* refSeq */ length,
+				clients.A.longClientId,
+				/* minSeq */ length,
+			),
+		);
+
+		// the below checks act as a proxy for the asymptotics of undo-redo
+		// linking. they are perhaps a bit more strict than necessary. if these
+		// tests are failing and the number of calls is still within a sane limit,
+		// it should be fine to update these checks to allow a larger number of
+		// calls
+		assert(
+			linkCount <= length * 2,
+			`expected tracking group link to occur at most twice per segment. found ${linkCount}`,
+		);
+		assert(
+			unlinkCount <= length,
+			`expected tracking group unlink to occur at most once per segment. found ${unlinkCount}`,
+		);
+
+		unspy1();
```

Comment:
the mock resetting won't happen on test failure. it's probably not a big deal for this test (spy functions are trivial), but might be worth writing a helper and wrapping in a try/finally block